icu4j/main/classes/core/src/com/ibm/icu/text/UnicodeSet.java

```diff
@@ -3766,7 +3779,7 @@ private UnicodeSet applyPropertyPattern(String pattern, ParsePosition ppos, Symb
         applyPropertyAlias(propName, valueName, symbols);
 
         if (invert) {
-            complement();
+            complement().removeAllStrings();  // code point complement
```

Comment:
Looking at the code, there could be 3 levels of complements here: applyPropertyPattern calls applyPropertyAlias which calls applyIntPropertyValue, all of which have a code path to `.complement()`.  Is there any possibility that information could be lost, e.g., by a double complement?

Reply:
I don't know what you mean. Each of these places should `.complement().removeAllStrings()` when the binary value is 0/false/no or when there is a `^`, so those strings are definitely "lost".

A code point complement of a code point complement is not a no-op, it removes the strings.